sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java

```diff
@@ -877,6 +879,34 @@ public CosmosClientBuilder sessionRetryOptions(SessionRetryOptions sessionRetryO
         return this;
     }
 
+    /**
+     * Sets the regions to exclude from the list of preferred regions. This means the request will not be
+     * routed to these excluded regions for non-retry and retry scenarios
+     * for the workload executed through this instance of {@link CosmosClient} / {@link CosmosAsyncClient}.
+     *
+     * @param excludeRegions The list of regions to exclude.
+     * @return current CosmosClientBuilder.
+     * */
+    public CosmosClientBuilder excludeRegions(List<String> excludeRegions) {
```

Comment:
hmm, do we need this, would it be enough to just set on each request

Reply:
It was a customer ask to configure a global / client-level `excludeRegions`.